tool/src/main/java/org/wildfly/security/tool/Command.java

```diff
@@ -270,4 +326,177 @@ protected Supplier<Provider[]> getProvidersSupplier(final String providersList)
+        String csType = PropertiesCredentialStore.NAME;
+        try {
+            credentialStore = CredentialStore.getInstance(csType);
+        } catch (NoSuchAlgorithmException e) {
+            // fallback to load all possible providers
+            credentialStore = CredentialStore.getInstance(csType, getProvidersSupplier(null));
+        }
+        Map<String, String> implProps = new HashMap<>();
+        implProps.put("create", String.valueOf(createCredentialStore));
+        implProps.put("location", credentialStoreLocation);
+        implProps.put("modifiable", Boolean.TRUE.toString());
+
+        try {
+            credentialStore.initialize(implProps);
+        } catch (CredentialStoreException e) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlockCredentialStoreNotLoaded(descriptorBlockCount));
+        }
+
+        SecretKeyCredential secretKeyCredential;
+        try {
+            secretKeyCredential = credentialStore.retrieve(alias, SecretKeyCredential.class);
+        } catch (UnsupportedCredentialTypeException e) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlockSecretKeyUnsupported(credentialStoreLocation, descriptorBlockCount));
+            return null;
+        }
+
+        // Acquire SecretKey, and populate credential store if set
+        SecretKey key;
+        if (secretKeyCredential != null) {
+            System.out.println(ElytronToolMessages.msg.existingCredentialStore());
+            key = secretKeyCredential.getSecretKey();
+        } else if (populate) {
+            try {
+                SecretKey newKey = SecretKeyUtil.generateSecretKey(256);
+                SecretKeyCredential newKeyCredential = new SecretKeyCredential(newKey);
+                credentialStore.store(alias, newKeyCredential);
+                credentialStore.flush();
+
+                key = newKey;
+            } catch (GeneralSecurityException e) {
+                warningHandler(ElytronToolMessages.msg.skippingDescriptorBlockUnableToPopulateCredentialStore(
+                        credentialStoreLocation, descriptorBlockCount));
+                return null;
+            }
+
+        } else {
+            warningHandler(ElytronToolMessages.msg.cmdFileSystemEncryptionNoSecretKey(credentialStoreLocation, descriptorBlockCount));
+            return null;
+        }
+
+        return key;
+    }
+
+    /**
+     * Acquire a given keypair from a {@link KeyStore}.
+     *
+     * @param keyPairAlias the name for a keypair within the keyStore
+     * @param descriptorBlockCount index of a descriptor block from bulk conversion files
+     * @return the requested {@link KeyPair}, or {@code null} if it could not be retrieved
+     */
+    KeyPair getKeyPair(Path keyStorePath, String keyStoreType, String keyPairAlias, char[] password, int descriptorBlockCount) throws Exception {
+        AtomicLoadKeyStore keyStore = null;
```

Comment:
Just to make sure, have the parameters been validated before this helper method is called? (e.g., keyPairAlias will be non-null, etc.)

Reply:
just replied to the other comment, I'll still update this one so it's consistent with the call from the integrity command, but otherwise it should be fine as is